### ISSUE TITLE
Code Style?

### ISSUE THREAD

**Issue by SirMoM** [KOBUGE-Games/godot-logger#12]
I think it would be a good idea to use a code style for this project.
It would make it easier to review and contribute.

I would propose using [gdscript-toolkit](https://github.com/Scony/godot-gdscript-toolkit), which includes a linter and a code formatter to create uniform code.

What are your thoughts on this topic @akien-mga @Samuel-Lewis?

**Comment by Samuel-Lewis** [comment]
Absolutely.

I'm a big advocate for consistency, and code style/linting is generally a big yes from me (see one of my [first commits](https://github.com/Samuel-Lewis/godot-logger/commit/e88276463993320289dfa4e5404c989ee408946f) on my fork :P).

Would be cool to also add some pre-commit hooks or some GitHub code actions to enforce it across the repo as well.

**Comment by Calinou** [comment]
I'd advise following the [GDScript style guide](https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_styleguide.html) :slightly_smiling_face: 

The formatter you linked can automate some of this, but parts of the style guide can't be enforced automatically yet.

**Comment by SirMoM** [comment]
So how to we go from here? What should be the result of this discussion? 
Use the formatter and review the code based on the "Style Guide"?
How can we enforce the usage of the formatter?

**Comment by Calinou** [comment]
> Use the formatter and review the code based on the "Style Guide"?

Sure, feel free to open a pull request for that :slightly_smiling_face: 

Since modifying code style creates a large diff, you'll most likely have to rebase it to solve merge conflicts before we can merge it (since other PRs will have higher priority).

> How can we enforce the usage of the formatter?

It's possible to run it in continuous integration (e.g. with GitHub Actions). I can't make a complete tutorial for this due to lack of time, but it should be feasible for anyone with some CI experience under their belt.

**Comment by akien-mga** [comment]
I agree on enforcing a consistent style. The linked GDScript style guide is good.

I don't have experience with gdscript-toolkit but of its output is compatible with the style guide, that sounds good to me.

And I agree about adding it to CI via GitHub Actions. You can look at actions on the Godot repo for an example (see static checks workflow).
